projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/case-edit/case-edit.component.ts

```diff
@@ -593,4 +603,24 @@ export class CaseEditComponent implements OnInit, OnDestroy {
     }
     return false;
   }
+
+  private taskIsForEvent(task: Task, eventDetails: EventDetails): boolean {
+    // EXUI-2668 - Ensure description for task includes event ID
+    // Note - This is a failsafe for an edge case that may never occur again
+    // Description may not include eventId in some cases which may mean task not completed (however this will be easy to check)
+    this.abstractConfig.logMessage(`checking taskIsForEvent: task ID ${task.id}, task description ${task.description}, event name ${eventDetails.eventId}`);
+    return task.case_id === eventDetails.caseId && (task.description && task.description.includes(eventDetails.eventId));
+  }
+
+  private removeTaskFromClientContext(): void {
```

Comment:
doesn't quite do what it says on the tin. I think I'd like it better if it did:
const foo = JSON.parse(this.sessionStorageService.getItem(CLIENT_CONTEXT))
if (foo?.client_context?.task_whatever) 
  delete foo.client_context.task_whatever